### PR TITLE
feat(lb): Add a callback for checking core node compatibility

### PR DESCRIPTION
**Pull request by ieQu1**

### CODE REVIEW

File: src/mria_lb.erl [hunk 1]
```diff
@@ -192,29 +193,55 @@ find_clusters(NodeInfo) ->
...
+        {ok, CustomCheckFun} ->
+            ok;
+        undefined ->
+            CustomCheckFun = fun(_) -> true end
+    end,
+    CustomInfo = maps:get(custom_info, LbInfo, undefined),
+    MyProtoVersion = mria_rlog:get_protocol_version(),
+    %% Actual check:
+    IsCustomCompat = try
+                         Result = CustomCheckFun(CustomInfo),
```
Comment (zmstone) [inline src/mria_lb.erl:231]: storing anonymous func here causes hot-path issues.
please stopre `{Function, Args}` instead so one can use a remote func for `Function`

File: src/mria_lb.erl [hunk 2]
```diff
@@ -287,11 +314,16 @@ lb_callback() ->
                   {ok, Vsn} -> Vsn;
                   undefined -> undefined
               end,
+    CustomInfo = case mria_config:callback(lb_custom_info) of
+                     {ok, CB}  -> CB();
```
Comment (thalesmg) [inline src/mria_lb.erl:318]: Should we wrap this in a `try` so it doesn't crash, similar to `CustomCheckFun/1`?

Comment (ieQu1) [reply src/mria_lb.erl:318]: This function is called via RPC, so if it crashes, it's equivalent to remote node being down.

File: src/mria_lb.erl (hunk 1 above)
Comment (ieQu1) [reply src/mria_lb.erl:231]: Good catch

**Review by zmstone — APPROVED** [review]

File: src/mria_lb.erl (hunk 1 above)
Comment (zmstone) [reply src/mria_lb.erl:231]: `CustomCheckFun` can already be a remote call.

**Pull request merged**